guide/canvas/README.md

```diff
@@ -0,0 +1,334 @@
+
+```js
+const Discord = require('discord.js');
+const Canvas = require('canvas');
+const snekfetch = require('snekfetch');
+
+client.on('guildMemberAdd', async member => {
+	const channel = member.guild.channels.find(ch => ch.name === 'member-log');
+	if (!channel) return;
+
+	const canvas = Canvas.createCanvas(700, 200);
+	const ctx = canvas.getContext('2d');
+
+	const background = await Canvas.loadImage('./wallpaper.jpg');
+	ctx.drawImage(background, 0, 0, canvas.width, canvas.height);
+
+	ctx.strokeStyle = '#FF0000';
+	ctx.strokeRect(0, 0, canvas.width, canvas.height);
+
+	// Pick up the pen
+	ctx.beginPath();
+	// Start the arc to form a circle
+	ctx.arc(125, 125, 100, 0, Math.PI * 2, true);
+	// Put the pen down
+	ctx.closePath();
+	// Needed for some reason
+	ctx.clip();
+
+	const buffer = await snekfetch.get(member.user.displayAvatarURL).then(r => r.body);
+	const avatar = await Canvas.loadImage(buffer);
+	ctx.drawImage(avatar, 25, 25, 200, 200);
+
+	const attachment = new Discord.Attachment(canvas.toBuffer(), 'image.png');
+
+	channel.send(`Welcome to the server, ${member}`, attachment);
+});
+```
+
+If followed successfully, the image should now look like this: ![Image](/assets/img/r6CiT3M.png)
+
+<p class="tip">You can read documentation on `context.arc()` [here](https://www.w3schools.com/tags/canvas_arc.asp).</p>
+
+Now, let's quickly go over adding text to your image. I'm sure that will be helpful to make clear the intent of this image, since currently it's just an avatar floating on a starry background that comes out of nowhere.
+
+```js
+client.on('guildMemberAdd', async member => {
+	const channel = member.guild.channels.find(ch => ch.name === 'member-log');
+	if (!channel) return;
+
+	const canvas = Canvas.createCanvas(700, 250);
+	const ctx = canvas.getContext('2d');
+
+	const background = await Canvas.loadImage('./wallpaper.jpg');
+	ctx.drawImage(background, 0, 0, canvas.width, canvas.height);
+
+	ctx.strokeStyle = '#FF0000';
+	ctx.strokeRect(0, 0, canvas.width, canvas.height);
+
+	// Select the font size and type from one of the natively available fonts
+	ctx.font = '50px sans-serif';
+	// Select the style that will be used to fill the text in
+	ctx.fillStyle = '#FFFFFF';
+	// Actually fill the text with a solid color
+	ctx.fillText(member.displayName, canvas.width / 2.5, canvas.height / 1.8);
+
+	ctx.beginPath();
+	ctx.arc(125, 125, 100, 0, Math.PI * 2, true);
+	ctx.closePath();
+	ctx.clip();
+
+	const buffer = await snekfetch.get(member.user.displayAvatarURL).then(r => r.body);
+	const avatar = await Canvas.loadImage(buffer);
+	ctx.drawImage(avatar, 25, 25, 200, 200);
+
+	const attachment = new Discord.Attachment(canvas.toBuffer(), 'image.png');
+
+	channel.send(`Welcome to the server, ${member}`, attachment);
+});
+```
+
+If followed successfully, the image should now look like this: ![Image](/assets/img/3rLGb1s.png)
```

Comment:
Instead of repeating "If followed successfully, ..." each time, I think the screenshot itself under the codeblock would be alright. Repeating the same phrase multiple times on the same page seems a bit meh, but that may be just me;